xtra-libp2p/src/dialer.rs

```diff
@@ -71,17 +74,14 @@ impl xtra::Actor for Actor {
             self.stop_with_error(e, ctx);
         }
 
-        // TODO: add connection logic
-        // tokio::time::sleep(CONNECTION_TIMEOUT).await;
-        //
-        // if !self.connected {
-        //     self.stop_with_error(
-        //         Error::Failed {
-        //             source: anyhow!("Connection timeout lapsed"),
-        //         },
-        //         ctx,
-        //     );
-        // }
+        let this = ctx.address().expect("self to be alive");
+
+        self.tasks.add(async move {
+            tokio::time::sleep(CONNECTION_TIMEOUT).await;
+            this.send(StopIfNotConnected)
+                .await
+                .expect("to deliver stop message");
```

Comment:
it's `StopIfNotConnected` message.... I wanted it to be different on purpose, because here we can't really expect "self to be alive" as it's spawned into a long-running task.